daemon/cluster/swarm.go

```diff
@@ -549,6 +549,7 @@ func validateAddr(addr string) (string, error) {
 	}
 	newaddr, err := opts.ParseTCPAddr(addr, defaultAddr)
 	if err != nil {
+		// TODO(thaJeztah) why are we ignoring the error here? Is this to allow "non-tcp" addresses?
```

Comment:
This is something to look into, so left a TODO (logic is a bit weird; if it's intentional, it should have a comment explaining why)